pkg/api/v1/conversion.go

```diff
@@ -480,10 +480,9 @@ func Convert_v1_PodTemplateSpec_To_api_PodTemplateSpec(in *PodTemplateSpec, out
 		in.Spec.InitContainers = values
 
 		// Call defaulters explicitly until annotations are removed
-		for i := range in.Spec.InitContainers {
-			c := &in.Spec.InitContainers[i]
-			SetDefaults_Container(c)
-		}
+		tmpPod := &Pod{Spec: PodSpec{InitContainers: values}}
+		SetObjectDefaults_Pod(tmpPod)
```

Comment:
that's actually the only special case I can find. I'd be ok with setting the tmpPod.Spec.HostNetwork as well, and calling it a day. @smarterclayton, what's the timeline on dropping support for initcontainer annotations altogether?